### PR TITLE
move load metadataSchema to build

### PR DESCRIPTION
**Pull request by mingwli**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4785)

### CODE REVIEW

File: src/docfx/build/Build.cs [hunk 1]
```diff
@@ -240,5 +242,16 @@ private static IReadOnlyList<Document> GetBuildScope(Docset docset, TableOfConte
 
             return buildScope;
         }
+
+        private static JsonSchema LoadMetadataSchema(Config config, string docsetPath, string fallbackDocsetPath)
```
Comment (yufeih) [inline src/docfx/build/Build.cs]: How about putting it to `MetadataProvider`?

**Comment by yufeih** [comment]
need #4775 to break the doubly linked chain between `Document` and `Docset`

File: src/docfx/build/metadata/MetadataProvider.cs [hunk 2]
```diff
@@ -46,6 +46,17 @@ public MetadataProvider(Docset docset, Cache cache)
             return _metadataCache.GetOrAdd(file, GetMetadataCore);
         }
 
+        public static JsonSchema LoadMetadataSchema(Config config, string docsetPath, string fallbackDocsetPath)
```
Comment (OsmondJiang) [inline src/docfx/build/metadata/MetadataProvider.cs]: Why not make metadataSchema as a property of MetadataProvider and expose it to TemplateEngine

**Comment by mingwli** [comment]
no longer needed after #4775

**Pull request closed without merging**